irqbalance.c

```diff
@@ -530,13 +530,17 @@ int init_socket()
 		addr.sun_family = AF_UNIX;
 		if (bind(socket_fd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
 			log(TO_ALL, LOG_WARNING, "Daemon couldn't be bound to the abstract socket, bailing out.\n");
+			close(socket_fd);
+			socket_fd = 0;
```

Comment:
theres nothing wrong with this, but you don't need to rewrite socket_fd to 0 on failure, as any subsequent call will just rewrite it again in the new call to init_socket.

Also, please recode this using a label pattern (like you did in the main routine), so that we don't have multiple close calls in this function.  It would be better to have a ret variable that gets set to 0 or 1 with a goto and a label at the function epiloge. Something like:
ret = 0
...
if(socket_fd < 0) {
    ret = 1
   goto fail
}
....
done:
     return ret
fail:
     close(socket_fd)
     goto done

Reply:
because socket_fd will be close in main(), so rewrite socket_fd to 0. so do not close socket_fd in init_socket() ?
thanks.

Reply:
I have update the patch, not close socket_fd in init_socket(), thanks.

Reply:
this is better, yes.  But if you go this route, there is another location in main (the check for core count < 2), where you can goto the fail label and do the same free operaion.

Reply:
ok, I will modify as your suggest.

Reply:
I have update the patch, thanks.